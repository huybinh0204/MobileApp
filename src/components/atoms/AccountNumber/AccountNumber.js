import PropTypes from 'prop-types';
import React from 'react';
import { TouchableOpacity } from 'react-native';
import { useSelector } from 'react-redux';
import strings from '_localization';
import { FundDirectDepositStoreSelectors } from '_store/pageStore/fundDirectDepositStore';
import { AccountTitle, AccountValue, Container, Item } from './AccountNumber.styles';

const AccountNumber = ({ onPressAccount, onPressRouting, showAccountNumber }) => {
  const { accountNumber, routingNumber } = useSelector(
    FundDirectDepositStoreSelectors.getRenderData
  );

  const formatAccountNumber = () => {
    if (showAccountNumber) {
      return accountNumber;
    }
    return accountNumber ? `${accountNumber?.slice(0, 3)} ••• •••` : '-';
  };

  const formatRoutingNumber = () => {
    return routingNumber
      ? `${routingNumber?.slice(0, 3)} ${routingNumber?.slice(3, 6)} ${routingNumber?.slice(6, 9)}`
      : '-';
  };

  return (
    <Container>
      <Item>
        <AccountTitle>{strings.routing}</AccountTitle>
        <TouchableOpacity
          testID="routingButton"
          disabled={!onPressRouting}
          onPress={() => onPressRouting({ routingNumber })}
        >
          <AccountValue>{formatRoutingNumber()}</AccountValue>
        </TouchableOpacity>
      </Item>
      <Item>
        <AccountTitle>{strings.account}</AccountTitle>
        <TouchableOpacity
          testID="accountButton"
          disabled={!onPressAccount}
          onPress={() => onPressAccount({ accountNumber })}
        >
          <AccountValue>{formatAccountNumber()}</AccountValue>
        </TouchableOpacity>
      </Item>
    </Container>
  );
};

AccountNumber.propTypes = {
  onPressRouting: PropTypes.func,
  onPressAccount: PropTypes.func,
  showAccountNumber: PropTypes.bool,
};

AccountNumber.defaultProps = {
  onPressRouting: null,
  onPressAccount: null,
  showAccountNumber: false,
};

export default AccountNumber;
