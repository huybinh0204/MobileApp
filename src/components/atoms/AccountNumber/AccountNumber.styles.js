import styled from '@emotion/native';
import BaseText from '_components/atoms/BaseText/BaseText';

export const Container = styled.View(({ theme }) => ({
  flexDirection: 'row',
  marginBottom: theme.spacing.xs,
}));

export const Item = styled.View({
  flex: 1,
  alignItems: 'center',
  justifyContent: 'center',
});

export const AccountTitle = styled(BaseText)(({ theme }) => ({
  ...theme.typography.tinyContentRegular,
  color: theme.colors.beta500,
}));

export const AccountValue = styled(BaseText)(({ theme }) => ({
  ...theme.typography.heading6,
  color: theme.colors.beta900,
}));
