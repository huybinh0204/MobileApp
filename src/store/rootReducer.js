import { combineReducers } from 'redux';
import actionWatchReducer from 'redux-action-watch/lib/reducer';
import { resettableReducer } from 'reduxsauce';
import { AccountReducer } from './account';
import { AuthenticationReducer } from './authentication';
import { CustomerReducer } from './customer';
import { FundDirectDepositStoreReducer } from './pageStore/fundDirectDepositStore';
import { RegisterReducer } from './register';
import { TransferReducer } from './transfer';

const resettable = resettableReducer('RESET');

export default combineReducers({
  account: resettable(AccountReducer),
  auth: resettable(AuthenticationReducer),
  customer: resettable(CustomerReducer),
  fundDirectDepositStoreReducer: resettable(FundDirectDepositStoreReducer),
  register: RegisterReducer,
  transfer: resettable(TransferReducer),
  watcher: actionWatchReducer, // Must be last
});
