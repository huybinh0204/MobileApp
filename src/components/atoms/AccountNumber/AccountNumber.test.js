import { render, waitFor } from '@testing-library/react-native';
import React from 'react';
import { createStore } from 'redux';
import rootReducer from '_store/rootReducer';
import ComponentWithProviders from '_test/mocks/ComponentWithProviders';
import AccountNumber from './AccountNumber';

describe('AccountNumber', () => {
  let mockStore;

  beforeEach(() => {
    mockStore = createStore(rootReducer, {
      fundDirectDepositStoreReducer: {
        renderData: {
          accountNumber: '728983733',
          routingNumber: '988765640',
        },
      },
    });
  });

  it('should match the snapshot when state is opt in and status is active', () => {
    const { toJSON } = render(
      <ComponentWithProviders component={AccountNumber} store={mockStore} />
    );

    expect(toJSON()).toMatchSnapshot();
  });

  it('should show account and routing numbers', async () => {
    const { getByText } = render(
      <ComponentWithProviders component={AccountNumber} store={mockStore} />
    );

    const accountNumber = getByText('728 ••• •••');
    const routingNumber = getByText('988 765 640');

    await waitFor(() => {
      expect(accountNumber).toBeTruthy();
      expect(routingNumber).toBeTruthy();
    });
  });

  it('should show completely the account number', async () => {
    const { getByText } = render(
      <ComponentWithProviders
        component={AccountNumber}
        store={mockStore}
        componentProps={{ showAccountNumber: true }}
      />
    );

    const accountNumber = getByText('728983733');

    await waitFor(() => {
      expect(accountNumber).toBeTruthy();
    });
  });
});
