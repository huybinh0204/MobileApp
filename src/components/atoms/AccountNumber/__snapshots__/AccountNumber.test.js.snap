// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountNumber should match the snapshot when state is opt in and status is active 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "rgb(242, 242, 242)",
            "flex": 1,
          },
          undefined,
        ]
      }
    >
      <RNSScreenContainer
        onLayout={[Function]}
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <RNSScreen
          activityState={2}
          collapsable={false}
          forwardedRef={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <View
            collapsable={false}
            style={
              Object {
                "opacity": 1,
              }
            }
          />
          <View
            accessibilityElementsHidden={false}
            closing={false}
            gestureVelocityImpact={0.3}
            importantForAccessibility="auto"
            onClose={[Function]}
            onGestureBegin={[Function]}
            onGestureCanceled={[Function]}
            onGestureEnd={[Function]}
            onOpen={[Function]}
            onTransition={[Function]}
            pointerEvents="box-none"
            style={
              Array [
                Object {
                  "display": "flex",
                  "overflow": undefined,
                },
                Object {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                },
              ]
            }
            transitionSpec={
              Object {
                "close": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
                "open": Object {
                  "animation": "spring",
                  "config": Object {
                    "damping": 500,
                    "mass": 3,
                    "overshootClamping": true,
                    "restDisplacementThreshold": 10,
                    "restSpeedThreshold": 10,
                    "stiffness": 1000,
                  },
                },
              }
            }
          >
            <View
              collapsable={false}
              pointerEvents="box-none"
              style={
                Object {
                  "flex": 1,
                  "marginTop": 0,
                }
              }
            >
              <View
                collapsable={false}
                forwardedRef={[Function]}
                needsOffscreenAlphaCompositing={false}
                onGestureHandlerEvent={[Function]}
                onGestureHandlerStateChange={[Function]}
                style={
                  Object {
                    "flex": 1,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "flex": 1,
                        "overflow": "hidden",
                      },
                      Array [
                        Object {
                          "backgroundColor": "rgb(242, 242, 242)",
                        },
                        undefined,
                      ],
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                        "flexDirection": "column-reverse",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <View
                        style={
                          Array [
                            Object {
                              "flexDirection": "row",
                              "marginBottom": 14.5,
                            },
                            undefined,
                          ]
                        }
                      >
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#5F6A77",
                                  "fontFamily": "Poppins",
                                  "fontSize": 21.5,
                                  "lineHeight": 30.5,
                                },
                                undefined,
                              ]
                            }
                          >
                            Routing
                          </Text>
                          <View
                            accessibilityState={
                              Object {
                                "disabled": true,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                            testID="routingButton"
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 29,
                                    "lineHeight": 40,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              988 765 640
                            </Text>
                          </View>
                        </View>
                        <View
                          style={
                            Array [
                              Object {
                                "alignItems": "center",
                                "flex": 1,
                                "justifyContent": "center",
                              },
                              undefined,
                            ]
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              Array [
                                Object {
                                  "color": "#5F6A77",
                                  "fontFamily": "Poppins",
                                  "fontSize": 21.5,
                                  "lineHeight": 30.5,
                                },
                                undefined,
                              ]
                            }
                          >
                            Account
                          </Text>
                          <View
                            accessibilityState={
                              Object {
                                "disabled": true,
                              }
                            }
                            accessible={true}
                            collapsable={false}
                            focusable={true}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "opacity": 1,
                              }
                            }
                            testID="accountButton"
                          >
                            <Text
                              allowFontScaling={false}
                              style={
                                Array [
                                  Object {
                                    "color": "#1F1D25",
                                    "fontFamily": "Poppins-SemiBold",
                                    "fontSize": 29,
                                    "lineHeight": 40,
                                  },
                                  undefined,
                                ]
                              }
                            >
                              728 ••• •••
                            </Text>
                          </View>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </RNSScreen>
      </RNSScreenContainer>
      <View
        collapsable={false}
        pointerEvents="box-none"
        style={
          Object {
            "height": 44,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
            "zIndex": 1,
          }
        }
      />
    </View>
  </RNCSafeAreaProvider>
</View>
`;
